src/DataModel/IData.ts

```diff
@@ -0,0 +1,6 @@
+import {IMetadata} from "./IMetadata";
+
+export interface IData extends Object
```

Comment:
> I wouldn't like to see developers create instances

I think the client wants resource representations. What if instead `IData` we used `IResource` as the common denominator?

And metadata? I think I like the idea. You wat to separate the controls from the actual representation, right? A place to put operations, etc? So what if it was called `IHypermedia` or `IControls` (or `IHypermediaControls`?). 

``` js
var resource = client.getResource('http://example.com');

resource.controls.getApiDocumentation();
resource.controls.getOperations();
resource.controls.getLinks();
```